src/mame/dynax/ddenlovr.cpp

```diff
@@ -10062,11 +10062,11 @@ void mmpanic_state::mmpanic(machine_config &config)
 	GENERIC_LATCH_8(config, m_soundlatch);
 	m_soundlatch->data_pending_callback().set_inputline(m_soundcpu, INPUT_LINE_NMI);
 
-	YM2413(config, "ym2413", 3579545).add_route(ALL_OUTPUTS, "mono", 0.80);
+	YM2413(config, "ym2413", XTAL(14'318'181) / 4).add_route(ALL_OUTPUTS, "mono", 0.80);
 
-	AY8910(config, "aysnd", 3579545).add_route(ALL_OUTPUTS, "mono", 0.30);
+	AY8910(config, "aysnd", XTAL(14'318'181) / 8).add_route(ALL_OUTPUTS, "mono", 0.30);
 
-	OKIM6295(config, m_oki, 1022720, okim6295_device::PIN7_HIGH); // clock frequency & pin 7 not verified
+	OKIM6295(config, m_oki, XTAL(14'318'181) / 14, okim6295_device::PIN7_HIGH); // pin 7 not verified
```

Comment:
Divide-by-14 is a bit of an odd factor.  Are you sure about this?

Reply:
Sorry. I'm not sure.
This change was referenced to Animalandia Jr. PCB Notes.
It has following description (= 1022727.21) and it almost matches previous clock value (= 1022720).
> M6295 clock- 1.02272MHz [14.31818/14]. Sample rate = 2000000 / 132

I began to feel like this should be `XTAL(16'000'000) / 16`, but both values suit PCB sound.
Should I revert here?

Reply:
14.31818/14 seems right to me. I don't know what the 2000000 in the notes means, but 132 means pin 7 is high.
It matches PCB notes and also other games on similar hw in this file which have this:
// clock frequency verified 1.022MHz, pin 7 verified high
OKIM6295(config, m_oki, XTAL(28'636'363)/28, okim6295_device::PIN7_HIGH);

Reply:
What’s it using to do the divide-by-seven though?  Was there some common support chip for doing the odd factor, or do they have counters and glue logic?

Reply:
You'd have to study the PCB for that. I'm sure it can be done with a couple TTL chips.

Reply:
The logic is non-trivial (not just a couple of chips, a mess of flip-flops) if you want a symmetric divide-by-seven.  If it’s doing a divide-by-eight and divide-by-28 on the same signal, it would only be able to share one divider stage at most without needing a complex symmetric divide-by-seven circuit.  Assuming it uses a 3-bit counter do do the divide-by-eight:
* It could tap the first output and do a simple divide-by-seven (another 3-bit counter plus some gates), and then feed that through another flip-flop for a symmetric divide by two.
* It could tap the second output and do a symmetric divide-by-seven (at least four flip-flops plus a more complex arrangement of gates).

Either way, it ends up with fairly messy logic, and not just “a couple of TTL chips”, as it needs multiple counter stages and/or flip-flops as well as combinatorial logic.  This is why odd factors are fairly unusual without dedicated support chips or ULAs.  I’m not saying it doesn’t happen, but it’s usually better to double-check things like this.

Reply:
Go for a simpler solution. A decimal counter chip. 3 outputs ANDed together to both reset the counter and clock the OKI. Something like that, who knows.

Reply:
That doesn’t give you a symmetric clock, and if you mean doing an asynchronous reset you’d get glitches which would be unacceptable in a clock generation circuit.  You wouldn’t actually make a clock generation circuit like that in practice.

Reply:
It really doesn't matter to MAME how they did it, though it makes much more sense to go for a simple solution.
The thing of note regarding this PR is that there are multiple PCB notes confirming the OKI input clock, enough to trust the odd divide factor of 28 or 14.